src/components/MapWindow/mapView.js

```diff
@@ -154,31 +159,41 @@ export default class MapWindow extends Component {
         map.invalidateSize();
         map.setView(MapApi.latLng(51.56, -0.06), 9);
         this.createCustomControls(map);
-
-
         const markers = new MapApi.MarkerClusterGroup();
         const markersList = [];
-        function populate(map) {
-            for (let i = 0; i < 1000; i++) {
+        function populate(map, count) {
+            for (let i = 0; i < count; i++) {
                 const m = new MapApi.Marker(getRandomLatLng(map));
                 markersList.push(m);
                 markers.addLayer(m);
             }
             return false;
         }
-       
+        const bounds = map.getBounds();
+
         function getRandomLatLng(map) {
-            const bounds = map.getBounds(),
-                southWest = bounds.getSouthWest(),
+            const southWest = bounds.getSouthWest(),
                 northEast = bounds.getNorthEast(),
                 lngSpan = northEast.lng - southWest.lng,
                 latSpan = northEast.lat - southWest.lat;
             return new MapApi.LatLng(
                 southWest.lat + latSpan * Math.random(),
                 southWest.lng + lngSpan * Math.random());
         }
-
-        populate(map);
+        const me = this;
+
+        function updateData() {
+            if (me.state.addCluster) {
+                const total = Math.random() * markers.lenght / 2;
```

Comment:
@Morphexe - I presume ````lenght```` is a typo. Since it doesn't exist, I'd expect it to be zero, so it wouldn't work.  So, it lenght actually a parameter?